content/using/running/vere.md

```diff
@@ -0,0 +1,633 @@
+details. Note that if your ship is behind a reverse proxy (nginx, Caddy, etc)
+and an SSL certificate is configured _there_, you'd not usually enable HTTPS on
+the ship itself.
+
+Note this is for secure HTTPS only, see `--http-port` for insecure HTTP.
+
+- Undocked: `urbit --https-port 8443 [pier]`
+- Docked: `[pier]/.run --https-port 8443`
+
+### `-I, --inject FILE`
+
+Inject the event stored in `FILE` jamfile.
+
+- Undocked: `urbit -I /path/to/file.jam [pier]`
+- Docked: `[pier]/.run -I /path/to/file.jam`
+
+### `-j, --json-trace`
+
+Create JSON trace files in `[pier]/.urb/put/trace`.
+
+- Undocked: `urbit -j [pier]`
+- Docked: `[pier]/.run urbit -j`
+
+### `-J, --ivory-pill PILL`
+
+This specifies an alternative Ivory pill to use. An Ivory pill is a pill
+compiled into the binary and used for runtime support code. You would not
+typically use this unless you're doing runtime or kernel development.
+
+- Undocked: `urbit -J /path/to/ivory.pill [pier]`
+- Docked: `[pier]/.run -J /path/to/ivory.pill`
+
+### `-k, --key-file KEYS`
+
+Boot a new ship with the private key file `KEY`. See `-G` for an alternative.
+
+- Undocked: `urbit -w sampel-palnet -k /path/to/my.key`
+
+### `-K, --kernel-stage STAGE`
+
+Start at Hoon kernel version stage specified. This option is currently a no-op,
+and doesn't do anything. It may be revived in the future or it may be removed.
+
+### `-l, --lite-boot`
+
+Most-minimal startup. This option is currently a no-op, and doesn't do anything.
+It may be revived in the future or it may be removed.
+
+### `-L, --local`
+
+Run a ship with local networking only. This will disable Ames networking, so the
+ship will not be able to communicate with other ships. HTTP(S) will continue to
+work as normal.
+
+- Undocked: `urbit -L [pier]`
+- Docked: `[pier]/.run -L`
+
+### `-n, --replay-to NUMBER`
+
+Replay up to the specified event `NUMBER`.
+
+- Undocked: `urbit -n 10000 [pier]`
+- Docked: `[pier]/.run -n 10000`
+
+A snapshot will be saved in a jamfile at `[pier]/.urb/roc/[NUMBER].jam`.
+
+### `--no-conn`
+
+Disable the control pane unix socket.
+
+- Undocked: `urbit --no-conn [pier]`
+- Docked: `[pier]/.run --no-conn`
+
+### `--no-dock`
+
+From v1.8 onwards, when a new ship is booted, the urbit binary will copy itself
+into the newly created pier by default (see the [`dock`](#dock) section for
+details). To prevent this from happening, the `--no-dock` option can be used.
+
+- Undocked: `urbit --no-dock -w sampel-palnet -k /path/to/my.key`
+
+### `-p, --ames-port PORT`
+
+Set the Ames UDP port to `PORT`. This is the port used for inter-ship
+communications.
+
+- Undocked: `urbit -p 33333 [pier]`
+- Docked: `[pier]/.run -p 33333`
+
+### `-P, --profile`
+
+Enable profiling. Profiles will be saved to
+`[pier]/.urb/put/profile/[now].txt`. This is useful for debugging and
+development. Profiling hit counters in Hoon code specified with the `~$` rune
+will be recorded here.
+
+- Undocked `urbit -P [pier]`
+- Docked `[pier]/.run -P`
```

Comment:
This runs a sampling profiler and produces custom output. Runtime cost metrics are only calculated if the binary was built with `CPU_DEBUG`.